tests/integration/test_grub.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+
+import pytest
+from charms.operator_libs_linux.v0 import grub
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(autouse=True)
+def clean_configs():
+    """Clean main and charms configs after each test."""
```

Comment:
I'm not totally up on pytest fixtures, but won't this run *before* each test? Do we instead want a [`yield` fixture](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#yield-fixtures-recommended) so that we can run code after the test?

```python
@pytest.fixture(autouse=True)
def clean_configs():
    yield  # run test
    # grub cleanup here
```